zeppelin-server/src/test/java/org/apache/zeppelin/socket/NotebookServerTest.java

```diff
@@ -412,6 +416,88 @@ public void testCreateNoteWithDefaultInterpreterId() throws IOException {
     notebook.removeNote(createdNote.getId(), anonymous);
   }
 
+  @Test
+  public void testHasCronExecutingUserUpdatePermission()
+          throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    Method method = NotebookServer.class.getDeclaredMethod("hasCronExecutingUserUpdatePermission",
```

Comment:
It is a too complicated to me to use reflection for the unit test of this trivial method that only compare 2 variables.

Reply:
@zjffdu Reflection is necessary to test the `hasCronExecutingUserUpdatePermission` method because it is a private method. Should I omit this `testHasCronExecutingUserUpdatePermission` test code to keep the code clean and simple?

Reply:
Usually private method is necessary to be tested, instead we should test the public method which use this private method.

Reply:
@zjffdu OK, thanks. In this case, I decided to remove this `testHasCronExecutingUserUpdatePermission` unit test because the test target method was so trivial at https://github.com/apache/zeppelin/pull/2671/commits/718e32955d3e6bca26aff0bd8b0a05f71669ac3a.